src/scripts/main.js

```diff
@@ -1,13 +1,30 @@
 'use strict';
 
+const form = document.querySelector('form');
+
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    if (element) {
+      element.addEventListener(eventName, () => {
+        const string = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
+
+        resolve(string);
+      });
+    } else{
+      reject(new Error('Element doesn\'t exist'));
```

Comment:
It's better to include more details in the error message, like which element is missing. This will make debugging easier.